manm_cs/noise/discrete_noise_builder.py

```diff
@@ -15,9 +15,14 @@ def with_num_discrete_values(self, num_discrete_values: int) -> 'DiscreteNoiseBu
         return self
 
     def build(self) -> DiscreteNoise:
+        # Portion transmitted to uniform noise, i.e., signal=0 for no noise and signal =1 for uniform noise
         signal = self.signal_to_noise_ratio
+        
+        # Portions of unfiform noise over all discrete values
+        signal_noise_classes = signal / self.num_discrete_values
```

Comment:
is it necessary to introduce `signal`, or could you directly work with `self.signal_to_noise_ratio` instead